webapp/lib/controller.dart

```diff
@@ -90,6 +102,61 @@ void initUI() {
   );
 }
 
+void setupProjectTimers() {
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+  for (var project in allProjects) {
+    var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+    projectTimers.add(new ProjectTimer(project, timer));
+  }
+}
+
+void setupProjectTimer(String project) {
+  var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+  projectTimers.add(new ProjectTimer(project, timer));
+}
+
+void checkNeedsReplyMetricsStale([bool isProjectSelection = false]) {
+  List<model.NeedsReplyData> sortedNeedsReplyDataList = List.from(needsReplyDataList);
+  sortedNeedsReplyDataList.sort((d1, d2) => d1.datetime.compareTo(d2.datetime));
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+
+  List<model.NeedsReplyData> latestDataPerProject = [];
+  for (var project in allProjects) {
+    var latestProject = sortedNeedsReplyDataList.where((entry) => entry.project == project).last;
+    latestDataPerProject.add(latestProject);
+  }
+
+  if (projectTimers.length < 1) {
+      setupProjectTimers();
+  } else {
+    for (var entry in latestDataPerProject) {
+      var projectTimer = projectTimers.firstWhere((projectTimer) => projectTimer.project == entry.project);
+      if (projectTimer != null) {
+        projectTimer.timer.cancel();
+        setupProjectTimer(entry.project);
+      }
+    }
+  }
+
+  var selectedProjectInLatest = latestDataPerProject.firstWhere((entry) =>
+      entry.project == view.contentView.projectSelectorView.selectedProject);
+  if (selectedProjectInLatest != null) {
+    if (isProjectSelection) {
```

Comment:
Considering whether we're in the project selection or not here complicates the logic a bit too much - consider splitting this method to tackle the data update case and the project selection separately.

Reply:
Simplified the logic.